store/driver/txn/scanner.go

```diff
@@ -32,3 +35,157 @@ func (s *tikvScanner) Next() error {
 func (s *tikvScanner) Key() kv.Key {
 	return kv.Key(s.Scanner.Key())
 }
+
+type oneByOneIter struct {
+	iters []kv.Iterator
+	cur   int
+}
+
+func newOneByOneIter(iters []kv.Iterator) *oneByOneIter {
+	iter := &oneByOneIter{
+		iters: iters,
+		cur:   0,
+	}
+
+	iter.updateCur()
+	return iter
+}
+
+func (o *oneByOneIter) updateCur() {
+	for o.cur >= 0 && o.cur < len(o.iters) {
+		if o.iters[o.cur].Valid() {
+			break
+		}
+		o.cur++
+	}
+}
+
+func (o *oneByOneIter) Valid() bool {
+	return o.cur >= 0 && o.cur < len(o.iters)
+}
+
+func (o *oneByOneIter) Next() error {
+	if !o.Valid() {
+		return errors.New("iterator is invalid")
+	}
+
+	err := o.iters[o.cur].Next()
+	if err != nil {
+		o.Close()
+		return err
+	}
```

Comment:
The design of this `oneByOneIter` is buggy.

When you close this one, will you call `oneByOneIter.Close()` later?
If not, it means resource leak, otherwise, it means close twice.
Does the iterator designed to be close-multiple-times safe?

In general, the resource management for iterator is difficult.
When you combine multiple iterators, the code become terrible to handle.